internal/bminventory/inventory_test.go

```diff
@@ -10029,6 +10029,160 @@ var _ = Describe("TestRegisterCluster", func() {
+					{
+						OpenShiftVersion:     swag.String("4.10.3"),
+						HighAvailabilityMode: swag.String("Full"),
+						DesiredNetworkType:   models.ClusterCreateParamsNetworkTypeOpenShiftSDN,
+						Platform:             models.Platform{Type: models.NewPlatformType(models.PlatformTypeBaremetal)},
+					},
+					// OpenShift version < 4.12, SNO => OVNKubernetes
+					{
+						OpenShiftVersion:     swag.String("4.11.17"),
+						HighAvailabilityMode: swag.String("None"),
+						DesiredNetworkType:   models.ClusterCreateParamsNetworkTypeOVNKubernetes,
+						Platform:             models.Platform{Type: models.NewPlatformType(models.PlatformTypeNone)},
+					},
+				}
+
+				for _, t := range tests {
+
+					mockClusterRegisterSuccess(true)
+					mockAMSSubscription(ctx)
+
+					NewClusterParams := getDefaultClusterCreateParams()
+					NewClusterParams.OpenshiftVersion = t.OpenShiftVersion
+					NewClusterParams.HighAvailabilityMode = t.HighAvailabilityMode
+					NewClusterParams.Platform = &t.Platform
+
+					reply := bm.V2RegisterCluster(ctx, installer.V2RegisterClusterParams{
+						NewClusterParams: NewClusterParams,
+					})
+
+					Expect(reply).Should(BeAssignableToTypeOf(installer.NewV2RegisterClusterCreated()))
+					actual := reply.(*installer.V2RegisterClusterCreated).Payload
+					Expect(swag.StringValue(actual.NetworkType)).To(Equal(t.DesiredNetworkType))
+				}
+			})
+
+			It("With Nil High Availability Value", func() {
+				tests := []clusterParamsForTest{
+					// OpenShift version >= 4.12, HighAvailability = nil => OVNKubernetes
+					{
+						OpenShiftVersion:     swag.String("4.13.2"),
+						HighAvailabilityMode: nil,
+						DesiredNetworkType:   models.ClusterCreateParamsNetworkTypeOVNKubernetes,
+						Platform:             models.Platform{Type: models.NewPlatformType(models.PlatformTypeBaremetal)},
+					},
+					// OpenShift version < 4.12, HighAvailability = nil => OpenShiftSDN
+					{
+						OpenShiftVersion:     swag.String("4.8.5"),
+						HighAvailabilityMode: nil,
+						DesiredNetworkType:   models.ClusterCreateParamsNetworkTypeOpenShiftSDN,
+						Platform:             models.Platform{Type: models.NewPlatformType(models.PlatformTypeBaremetal)},
+					},
+				}
+
+				for _, t := range tests {
+
+					mockClusterRegisterSuccess(true)
+					mockAMSSubscription(ctx)
+
+					NewClusterParams := getDefaultClusterCreateParams()
+					NewClusterParams.OpenshiftVersion = t.OpenShiftVersion
+					NewClusterParams.HighAvailabilityMode = t.HighAvailabilityMode
+					NewClusterParams.Platform = &t.Platform
+
+					reply := bm.V2RegisterCluster(ctx, installer.V2RegisterClusterParams{
+						NewClusterParams: NewClusterParams,
+					})
+
+					Expect(reply).Should(BeAssignableToTypeOf(installer.NewV2RegisterClusterCreated()))
+					actual := reply.(*installer.V2RegisterClusterCreated).Payload
+					Expect(swag.StringValue(actual.NetworkType)).To(Equal(t.DesiredNetworkType))
+				}
```

Comment:
it seems to be the same code as above cases
why not merge ``availability-mode=nil`` to the above dictionary cases?

Reply:
same question for the pre-release cases

Reply:
I wanted to create a semantic separation between the tests, thought it would be too long in one test